src/Test/Utilities/Shared/Compilation/TestOperationWalker.cs

```diff
@@ -0,0 +1,541 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.Semantics;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Test.Utilities
+{
+    public sealed class TestOperationWalker : OperationWalker
+    {
+        private static TestOperationWalker s_instance;
+
+        private TestOperationWalker()
+            : base()
+        { }
+
+        public static TestOperationWalker GetInstance()
+        {
+            if (s_instance == null)
+            {
+                s_instance = new TestOperationWalker();
+            }
+            return s_instance;
+        }
+
+#if TEST_NONE_OPERATION
+        internal override void VisitNoneOperation(IOperation operation)
+        {
+            Assert.True(false, "Encountered an IOperation with `Kind == OperationKind.None` while walking the operation tree.");
+        }
+#endif
```

Comment:
This override will throw when a None kind operation is encountered, so needs to be enabled separately.